pkg/drivers/nats/server/server.go

```diff
@@ -0,0 +1,36 @@
+//go:build nats
+// +build nats
+
+package server
+
+import (
+	"fmt"
+
+	"github.com/nats-io/nats-server/v2/server"
+)
+
+func New(configFile string, dontListen, stdoutLogging bool) (Server, error) {
+	opts := &server.Options{}
+
+	if configFile == "" {
+		// TODO: Other defaults for easy single node config?
+		opts.JetStream = true
+	} else {
+		// Parse the server config file as options.
+		var err error
+		opts, err = server.ProcessConfigFile(configFile)
+		if err != nil {
+			return nil, fmt.Errorf("failed to process NATS server config file: %w", err)
+		}
+	}
+
+	opts.DontListen = dontListen
+	opts.Port = 0
```

Comment:
How do the DontListen and Port options interact? If you don't set the DontListen option, does it listen on a random port, or is there a default is uses instead?

Reply:
That was an oversight, thanks. Port does not need to be reset if defined in the config. Internally if `DontListen` is set, `Port` is ignored. Fixed.